chassis/core/src/main/java/com/griddynamics/jagger/engine/e1/process/AbstractWorkloadService.java

```diff
@@ -0,0 +1,407 @@
+
+    public void changeDelay(int delay) {
+        this.delay.set(delay);
+    }
+
+    public static class WorkloadServiceBuilder {
+        private final Scenario<Object, Object, Object> scenario;
+        private Executor executor = NewThreadExecutor.INSTANCE;
+        private ImmutableList.Builder<ScenarioCollector<?, ?, ?>> collectors = ImmutableList.builder();
+        private ImmutableList.Builder<Validator> validators = ImmutableList.builder();
+        private ImmutableList.Builder<InvocationListener> listeners = ImmutableList.builder();
+
+        private WorkloadServiceBuilder(Scenario<Object, Object, Object> scenario) {
+            this.scenario = scenario;
+        }
+
+        public WorkloadServiceBuilder addCollectors(List<ScenarioCollector<?, ?, ?>> collectors) {
+            this.collectors.addAll(collectors);
+            return this;
+        }
+
+        public WorkloadServiceBuilder addCollector(ScenarioCollector<?, ?, ?> collector) {
+            collectors.add(collector);
+            return this;
+        }
+
+        public WorkloadServiceBuilder addValidators(List<Validator> validators) {
+            this.validators.addAll(validators);
+            return this;
+        }
+
+        public WorkloadServiceBuilder addListeners(List<InvocationListener<?, ?, ?>> listeners) {
+            this.listeners.addAll(listeners);
+            return this;
+        }
+
+        public WorkloadServiceBuilder useExecutor(Executor executor) {
+            this.executor = executor;
+            return this;
+        }
+
+        public WorkloadService buildInfiniteService() {
+            ImmutableList<ScenarioCollector<?, ?, ?>> collectorsList = collectors.build();
+            ImmutableList<Validator> validatorList = validators.build();
+            ImmutableList<InvocationListener> listenersList = listeners.build();
+
+            scenario.setInvocationListener(Invokers.validateListener(validatorList, (Iterable) collectorsList, (List)listenersList));
+            return new InfiniteWorkloadService(Invokers.mergeFlushElements(validatorList, collectorsList));
+        }
+
+        public WorkloadService buildServiceWithPredefinedSamples(final int samples) {
+            ImmutableList<ScenarioCollector<?, ?, ?>> collectorsList = collectors.build();
+            ImmutableList<Validator> validatorList = validators.build();
+            ImmutableList<InvocationListener> listenersList = listeners.build();
+
+            scenario.setInvocationListener(Invokers.validateListener(validatorList, (Iterable) collectorsList, (List)listenersList));
+            return new PredefinedSamplesWorkloadService(Invokers.mergeFlushElements(validatorList, collectorsList), samples);
+        }
+
+        public WorkloadService buildServiceWithSharedSamplesCount(final AtomicInteger samples) {
+            ImmutableList<ScenarioCollector<?, ?, ?>> collectorsList = collectors.build();
+            ImmutableList<Validator> validatorList = validators.build();
```

Comment:
Yes, but i did not refactor all logic of WorkloadServiceCreation. It moved here from WorkloadService class.